### ISSUE TITLE
Linter doesn't detect SELECT statements with missing FROM

### ISSUE THREAD

**Issue by apamildner** [sqlfluff/sqlfluff#6245]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

If we have a statement like this
```sql
SELECT (a, b);

```
the linter should complain that we are missing a FROM statement, as this is not valid if I understand the SQL spec correctly.

### Expected Behaviour

The linter should error out when we are missing a FROM statement in a SELECT.

### Observed Behaviour

The linter completes without errors.

### How to reproduce

Create a file `example.sql`
```sql
SELECT (a, b);

```
Then run
`sqlfluff --dialect databricks example.sql` 

### Dialect

I tried with `databricks`, `postgres`

### Version

`sqlfluff: version 3.2.0`
`Python: 3.12.6`

### Configuration

No config set

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by greg-finley** [comment]
Thanks for reaching out. However, our goal is to format your SQL, not ensure that the SQL is actually valid. It's expected behavior that we can lint some invalid stuff as well.

**Comment by apamildner** [comment]
OK thanks for the quick response!
